src/Analyzers/MSTest.Analyzers/DataRowShouldBeValidAnalyzer.cs

```diff
@@ -0,0 +1,172 @@
+        context.EnableConcurrentExecution();
+
+        context.RegisterCompilationStartAction(context =>
+        {
+            // No need to register any action if we don't find the TestMethodAttribute symbol since
+            // the current analyzer checks if the DataRow attribute is applied on test methods. No
+            // test methods, nothing to check.
+            if (!context.Compilation.TryGetOrCreateTypeByMetadataName(
+                WellKnownTypeNames.MicrosoftVisualStudioTestToolsUnitTestingTestMethodAttribute,
+                out var testMethodAttributeSymbol))
+            {
+                return;
+            }
+
+            if (!context.Compilation.TryGetOrCreateTypeByMetadataName(
+                WellKnownTypeNames.MicrosoftVisualStudioTestToolsUnitTestingDataRowAttribute,
+                out var dataRowAttributeSymbol))
+            {
+                return;
+            }
+
+            context.RegisterSymbolAction(
+                context => AnalyzeSymbol(context, testMethodAttributeSymbol, dataRowAttributeSymbol),
+                SymbolKind.Method);
+        });
+    }
+
+    private static void AnalyzeSymbol(
+        SymbolAnalysisContext context,
+        INamedTypeSymbol testMethodAttributeSymbol,
+        INamedTypeSymbol dataRowAttributeSymbol)
+    {
+        IMethodSymbol methodSymbol = (IMethodSymbol)context.Symbol;
+
+        bool isTestMethod = false;
+        List<AttributeData> dataRowAttributes = new();
+        foreach (var methodAttribute in methodSymbol.GetAttributes())
+        {
+            // Current method should be a test method or should inherit from the TestMethod attribute.
+            // If it is, the current analyzer will trigger no diagnostic so it exits.
+            if (methodAttribute.AttributeClass.Inherits(testMethodAttributeSymbol))
+            {
+                isTestMethod = true;
+            }
+
+            if (SymbolEqualityComparer.Default.Equals(methodAttribute.AttributeClass, dataRowAttributeSymbol))
+            {
+                dataRowAttributes.Add(methodAttribute);
+            }
+        }
+
+        // Check if attribute is set on a test method.
+        if (!isTestMethod)
+        {
+            if (dataRowAttributes.Count > 0)
+            {
+                context.ReportDiagnostic(methodSymbol.CreateDiagnostic(DataRowOnTestMethodRule));
+            }
+
+            return;
+        }
+
+        // Check each data row attribute.
+        foreach (var attribute in dataRowAttributes)
+        {
+            AnalyzeAttribute(context, attribute, methodSymbol);
+        }
+    }
+
+    private static void AnalyzeAttribute(SymbolAnalysisContext context, AttributeData attribute, IMethodSymbol methodSymbol)
+    {
+        if (attribute.ApplicationSyntaxReference?.GetSyntax() is not { } syntax)
+        {
+            return;
+        }
+
+        if (attribute.ConstructorArguments.Length == 0)
```

Comment:
I would argue that breaking is desired in this case. The fix is very low effort and we already agreed that using an empty `DataRow` makes no sense so this will declutter the test code. This will also ease the way to deprecating the no args constructor.

Reply:
Removed this case, thanks.